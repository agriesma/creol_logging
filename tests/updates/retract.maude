load creol-update .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> State [ctor] .
  eq state = {
    remove("BankAccount", (noInh), (noSubst),
      (< "balance" : Method | Param: (noVid), Att: (noSubst),
         Code: skip >), (c("BankAccount", 0)), (none))

    < class("BankAccount", 0) : Class | Version: 0, Inh: noInh, Param: noVid,
      Att: "bal" |-> null,
      Mtds: < "balance" : Method | Param: noVid, Att: "_" |->
              null, "b" |-> null, Code: assign( "b" ; "bal" ) ;
              return ( "b" ) >,
            < "init" : Method | Param: noVid, Att: "_" |-> null,
              Code: assign( ( "bal" @ "BankAccount" ) ; int(0) ) ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "_" |-> null,
              Code: skip ; return ( emp ) >,
      Ocnt: 0 > } .
endm

