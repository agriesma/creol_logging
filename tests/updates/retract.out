==========================================
rewrite in PROGRAM : state .
result State: {
< class("BankAccount",1) : Class | Version: 1, Inh: noInh, Param: noVid, Att:
    "bal" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: assign("bal" @
    "BankAccount" ; int(0)) ; return(emp) >, < "run" : Method | Param: noVid,
    Att: "_" |-> null, Code: skip ; return(emp) >,
  Ocnt: 0 >
}
Bye.
